R/ggs_volcano.R

```diff
@@ -0,0 +1,181 @@
+    dds <- gtl$dds
+    res_de <- gtl$res_de
+    res_enrich <- gtl$res_enrich
+    annotation_obj <- gtl$annotation_obj
+  }
+
+  # Retrieve information about genes in geneset gs_id
+  if (!is.null(geneset_id)) {
+    if (geneset_id %in% res_enrich[["gs_id"]]) {
+      thisset_name <- res_enrich[geneset_id, "gs_description"]
+      thisset_members <- unlist(strsplit(res_enrich[geneset_id, "gs_genes"], ","))
+      thisset_members_ids <- annotation_obj$gene_id[match(thisset_members, annotation_obj$gene_name)]
+    }
+  }else {
+    # overwritable via a list
+    if (!all(genelist %in% rownames(res_de))) {
+      not_there <- genelist[!(genelist %in% rownames(res_de))]
+      warning("Some of the provided gene ids were not found in the SummarizedExperiment",
+              "\nNot found: ", not_there)
+    }
+    thisset_members_ids <- intersect(rownames(res_de), genelist)
+    thisset_name <- "Custom list"
+  }
+
+  thisset_members_ids
+  
+  # Prepare the data
+  complete_genes_ids <- rownames(res_de)
+  complete_genes <- annotation_obj$gene_name[match(complete_genes_ids, annotation_obj$gene_id)]
+  
+  padj_complete <- res_de[complete_genes_ids, "padj"]
+  filter_info_complete <- sapply(padj_complete, function(x) x <= FDR)
+  padj_complete <- sapply(padj_complete, function(x) -log10(x))
+  
+  log2FoldChange_complete <- res_de[complete_genes_ids, "log2FoldChange"]
+  
+  gene_set_belong <- complete_genes_ids %in% thisset_members_ids
+  filter_info_complete <- filter_info_complete & gene_set_belong
+
+  
+  thisset_complete_data <- data.frame(complete_genes_ids, padj_complete, log2FoldChange_complete, filter_info_complete, gene_set_belong)
+  colnames(thisset_complete_data) <- c("genes", "logTransformedpvalue", "log2FoldChange", "significant", "belonging")
+  
+  
+  # Prepare plotting
+  volcano_df_complete <- thisset_complete_data
+  volcano_df_complete$genes_name <- complete_genes
+  max_x <- max(abs(range(thisset_complete_data["log2FoldChange"])))
+  limit_x <- max_x * c(-1, 1)
+  
+
+  # Prepare plot title
+  if (is.null(plot_title)) {
+    title <- paste0("Volcano Plot - ", thisset_name, " - ", geneset_id)
+  } else {
+    title <- plot_title
+  }
+
+
+  # Plot data
+  p <- ggplot(volcano_df_complete, aes_string(x = "log2FoldChange", y = "logTransformedpvalue")) +
+       geom_point(aes_string(color = "significant", alpha = "belonging")) +
+       labs(x = "log2Fold Change",
+            y = "-log10 p-value",
+            color = "p-value <= FDR") + 
+       scale_x_continuous(limits = limit_x) +
+       scale_color_manual(labels = c("significant", "not significant"), 
+                          breaks = c("TRUE", "FALSE"),
+                          values = c(color, "grey25")) +
+       scale_alpha_manual(breaks = c("TRUE", "FALSE"),
+                          values = c(1, 1/10)) +
+       theme_bw() +
+       theme(legend.title = element_text(size = 11, face = "bold"),
+             legend.text = element_text(size = 10))  
```

Comment:
this one can be indented like the one below (I am an avid user of Cmd + I on Mac 😉 )